roles/heat/tasks/main.yml

```diff
@@ -13,47 +13,52 @@
   file: dest=/var/cache/heat state=directory mode=0700
         owner=heat group=heat
 
-- name: permit access to heat
-  ufw: rule=allow to_port={{ item }} proto=tcp
-  tags: ufw
-  with_items:
-    - "{{ endpoints.heat.port.haproxy_api }}"
-    - "{{ endpoints.heat_cfn.port.haproxy_api }}"
-  when: os == 'ubuntu'
-
-- name: Permit access to heat
-  firewalld:
-    state: enabled
-    permanent: true
-    immediate: true
-    port: "{{ item }}/tcp"
-  with_items:
-    - "{{ endpoints.heat.port.haproxy_api }}"
-    - "{{ endpoints.heat_cfn.port.backend_api }}"
-  tags: firewall
-  when: os == 'rhel'
+- block:
+  - name: permit access to heat
+    ufw: rule=allow to_port={{ item }} proto=tcp
+    tags: ufw
+    with_items:
+      - "{{ endpoints.heat.port.haproxy_api }}"
+      - "{{ endpoints.heat_cfn.port.haproxy_api }}"
 
-- name: install heat services
-  upstart_service: name={{ item }}
-                   user=heat
-                   cmd=/usr/local/bin/{{ item }}
-                   config_files=/etc/heat/heat.conf
-  with_items:
-    - heat-api
-    - heat-api-cfn
-    - heat-engine
+  - name: install heat services (ubuntu)
+    upstart_service:
+      name: "{{ item.name }}"
+      user: "{{ item.user }}"
+      cmd: "{{ item.cmd }}"
+    with_items:
+      - "{{ heat.services.heat_api }}"
+      - "{{ heat.services.heat_api_cfn }}"
+      - "{{ heat.services.heat_engine }}"
   when: os == 'ubuntu'
 
-- name: install heat services (rhel source)
-  systemd_service: name={{ item }}
-                   user=heat
-                   cmd=/usr/local/bin/{{ item }}
-                   config_files=/etc/heat/heat.conf
-  with_items:
-    - heat-api
-    - heat-api-cfn
-    - heat-engine
-  when: os == 'rhel' and openstack_install_method == 'source'
+- block:
+  - name: Permit access to heat
+    firewalld:
+      state: enabled
+      permanent: true
+      immediate: true
+      port: "{{ item }}/tcp"
+    with_items:
+      - "{{ endpoints.heat.port.haproxy_api }}"
+      - "{{ endpoints.heat_cfn.port.backend_api }}"
+    tags: firewall
+
+  - name: install heat services (rhel)
+    systemd_service:
+      name: "{{ item.name }}"
+      description: "{{ item.desc }}"
+      type: "{{ item.type }}"
+      user: "{{ item.user }}"
+      cmd: "{{ item.cmd }}"
+      config_dirs: "{{ item.config_dirs }}"
+      config_files: "{{ item.config_files }}"
+      restart: "{{ item.restart }}"
+    with_items:
+      - "{{ heat.services.heat_api }}"
```

Comment:
thanks for your comments. I forgot to refine start/stop services with new variables. Updated them and please review again.